File: Server/src/server/DatabaseAdapter.java
package server;

import java.util.List;

/**
 * 
 * @author Alberto Pareja-Lecaros
 *
 * The DatabaseAdapter acts as a bridge between the server code and the database.
 * 
 */
public interface DatabaseAdapter {

	public void storeIndexInDatabase(final String username, final String filename, final String deviceName);
	public boolean login(final String username, final String passwordHash);
	public List<Object[]> selectAllFilesFromUser(String username);
}

File: Server/src/main/scala/server/DatabaseAdapter.scala
package server
import file.UbiquityFileData

/**
 * The DatabaseAdapter acts as a bridge between
 * the server code and the database.
 */
trait DatabaseAdapter {

  def storeIndexInDatabase(username: String, filename: String,
    deviceName: String)
  def storeFileInDatabase(userid: Int, filename: String, filepath: String, filesize: Long, devicename: String)
  def getAllFilesOfType(userid: Int, fileid: Int): Array[UbiquityFileData];
  def login(username: String, passwordHash: String): Int
  def selectAllFilesFromUser(username: String): Vector[Array[String]]
  def userExists(username: String): Boolean
  def betaSignup(username: String, email: String): Boolean
  def getFileSize(username: String, filepath: String,
    deviceName: String): Long
  def insertDevice(devicename: String, macaddr: String, userid: Int)
  def getDevices(userid: Int): Array[String]
}
